client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java

```diff
@@ -54,15 +59,63 @@ public void testDeleteIndex() throws IOException {
         }
     }
 
+    public void testOpenExistingIndex() throws IOException {
+        String[] indices = randomIndices(1, 5);
+        for (String index : indices) {
+            createIndex(index);
+            closeIndex(index);
+            ResponseException exception = expectThrows(ResponseException.class, () -> client().performRequest("GET", index + "/_search"));
+            assertThat(exception.getResponse().getStatusLine().getStatusCode(), equalTo(RestStatus.BAD_REQUEST.getStatus()));
+            assertThat(exception.getMessage().contains(index), equalTo(true));
+        }
+
+        OpenIndexRequest openIndexRequest = new OpenIndexRequest(indices);
+        OpenIndexResponse openIndexResponse = execute(openIndexRequest, highLevelClient().indices()::openIndex,
+                highLevelClient().indices()::openIndexAsync);
+        assertTrue(openIndexResponse.isAcknowledged());
+
+        for (String index : indices) {
+            client().performRequest("GET", index + "/_search");
+        }
+    }
+
+    public void testOpenNonExistentIndex() throws IOException {
+        String[] nonExistentIndices = randomIndices();
+        for (String nonExistentIndex : nonExistentIndices) {
+            assertFalse(indexExists(nonExistentIndex));
+        }
+
+        OpenIndexRequest openIndexRequest = new OpenIndexRequest(nonExistentIndices);
+        ElasticsearchException exception = expectThrows(ElasticsearchException.class,
+                () -> execute(openIndexRequest, highLevelClient().indices()::openIndex, highLevelClient().indices()::openIndexAsync));
+        assertEquals(RestStatus.NOT_FOUND, exception.status());
```

Comment:
would you mind checking also that with `IndicesOptions.lenientExpandOpen` we don't get back an error?